pkg/build/controller/buildconfig/buildconfig_controller.go

```diff
@@ -136,7 +137,11 @@ func (c *BuildConfigController) handleBuildConfig(bc *buildapi.BuildConfig) erro
 		} else if buildgenerator.IsFatal(err) || kerrors.IsNotFound(err) || kerrors.IsBadRequest(err) || kerrors.IsForbidden(err) {
 			instantiateErr = fmt.Errorf("gave up on Build for BuildConfig %s due to fatal error: %v", bcDesc(bc), err)
 			utilruntime.HandleError(instantiateErr)
-			c.recorder.Event(bc, kapi.EventTypeWarning, "BuildConfigInstantiateFailed", instantiateErr.Error())
+			// Fixes https://github.com/openshift/origin/issues/16557
+			// Caused by a race condition between the ImageChangeTrigger vs the BuildConfigChangeTrigger
+			if !strings.Contains(instantiateErr.Error(), fmt.Sprintf("the LastVersion(1) on build config %s/%s does not match the build request LastVersion(0)", bc.Namespace, bc.Name)) {
+				c.recorder.Event(bc, kapi.EventTypeWarning, "BuildConfigInstantiateFailed", instantiateErr.Error())
+			}
```

Comment:
What if a single request comes through (i.e. not a race between ICT and CCT) after the initial BC create with a bad last version?  Wouldn't we be masking a legitimate error in that case?

Unfortunately, don't know if I have a super great solution ... perhaps we fetch the BC before performing this filter and see if it was *recently created* ... where determining what is *recent* is perhaps kludgy ... but maybe at least an improvement.

Dumb non-transactional databases

Reply:
> What if a single request comes through (i.e. not a race between ICT and CCT) after the initial BC create with a bad last version? 

request comes from where? to where?  this logic is all in the buildconfig controller.  There are no other buildrequests its going to process.

@coreydaley your check on LastVersion(1) is too specific.  I could potentially start multiple builds before the buildconfig controller puts in its request.  Then the error would be "the LastVersion(5) on build config..." or whatever.

Just check for the back half: `does not match the build request LastVersion(0)`